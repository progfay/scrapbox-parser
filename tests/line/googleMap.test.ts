/* global describe it expect */

describe('googleMap', () => {
  it('Simple google map with NE', () => {
    expect('[N35.6812362,E139.7649361]').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'googleMap',
            latitude: 35.6812362,
            longitude: 139.7649361,
            zoom: 14,
            place: '',
            url: 'https://www.google.com/maps/@35.6812362,139.7649361,14z'
          }
        ]
      }
    ])
  })

  it('Simple google map with SW', () => {
    expect('[S13.70533,W69.6533372]').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'googleMap',
            latitude: -13.70533,
            longitude: -69.6533372,
            zoom: 14,
            place: '',
            url: 'https://www.google.com/maps/@-13.70533,-69.6533372,14z'
          }
        ]
      }
    ])
  })

  it('Simple google map with zoom', () => {
    expect('[N35.6812362,E139.7649361,Z14]').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'googleMap',
            latitude: 35.6812362,
            longitude: 139.7649361,
            zoom: 14,
            place: '',
            url: 'https://www.google.com/maps/@35.6812362,139.7649361,14z'
          }
        ]
      }
    ])
  })

  it('Simple google map with place on left', () => {
    expect('[東京駅 N35.6812362,E139.7649361,Z14]').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'googleMap',
            latitude: 35.6812362,
            longitude: 139.7649361,
            zoom: 14,
            place: '東京駅',
            url: 'https://www.google.com/maps/place/%E6%9D%B1%E4%BA%AC%E9%A7%85/@35.6812362,139.7649361,14z'
          }
        ]
      }
    ])
  })

  it('Simple google map with place on right', () => {
    expect('[N35.6812362,E139.7649361,Z14 東京駅]').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'googleMap',
            latitude: 35.6812362,
            longitude: 139.7649361,
            zoom: 14,
            place: '東京駅',
            url: 'https://www.google.com/maps/place/%E6%9D%B1%E4%BA%AC%E9%A7%85/@35.6812362,139.7649361,14z'
          }
        ]
      }
    ])
  })
})
