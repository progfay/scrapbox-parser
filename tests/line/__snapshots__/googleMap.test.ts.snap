// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`googleMap Simple google map with NE 1`] = `
[
  {
    "indent": 0,
    "nodes": [
      {
        "latitude": 35.6812362,
        "longitude": 139.7649361,
        "place": "",
        "raw": "[N35.6812362,E139.7649361]",
        "type": "googleMap",
        "url": "https://www.google.com/maps/@35.6812362,139.7649361,14z",
        "zoom": 14,
      },
    ],
    "type": "line",
  },
]
`;

exports[`googleMap Simple google map with SW 1`] = `
[
  {
    "indent": 0,
    "nodes": [
      {
        "latitude": -13.70533,
        "longitude": -69.6533372,
        "place": "",
        "raw": "[S13.70533,W69.6533372]",
        "type": "googleMap",
        "url": "https://www.google.com/maps/@-13.70533,-69.6533372,14z",
        "zoom": 14,
      },
    ],
    "type": "line",
  },
]
`;

exports[`googleMap Simple google map with place on left 1`] = `
[
  {
    "indent": 0,
    "nodes": [
      {
        "latitude": 35.6812362,
        "longitude": 139.7649361,
        "place": "東京駅",
        "raw": "[東京駅 N35.6812362,E139.7649361,Z14]",
        "type": "googleMap",
        "url": "https://www.google.com/maps/place/%E6%9D%B1%E4%BA%AC%E9%A7%85/@35.6812362,139.7649361,14z",
        "zoom": 14,
      },
    ],
    "type": "line",
  },
]
`;

exports[`googleMap Simple google map with place on right 1`] = `
[
  {
    "indent": 0,
    "nodes": [
      {
        "latitude": 35.6812362,
        "longitude": 139.7649361,
        "place": "東京駅",
        "raw": "[N35.6812362,E139.7649361,Z14 東京駅]",
        "type": "googleMap",
        "url": "https://www.google.com/maps/place/%E6%9D%B1%E4%BA%AC%E9%A7%85/@35.6812362,139.7649361,14z",
        "zoom": 14,
      },
    ],
    "type": "line",
  },
]
`;

exports[`googleMap Simple google map with zoom 1`] = `
[
  {
    "indent": 0,
    "nodes": [
      {
        "latitude": 35.6812362,
        "longitude": 139.7649361,
        "place": "",
        "raw": "[N35.6812362,E139.7649361,Z14]",
        "type": "googleMap",
        "url": "https://www.google.com/maps/@35.6812362,139.7649361,14z",
        "zoom": 14,
      },
    ],
    "type": "line",
  },
]
`;
